### PR TITLE
Use coreclr buildtools nuget package instead of compiling the tools u…

### PR DESCRIPTION
**Pull request by rahku**
…sed during build.

This only applies to windows build.

### CODE REVIEW

**Comment by rahku** [comment]
@gkhanna79 @AlexGhiondea 
Can you please review

**Comment by janvorli** [comment]
@rahku What is the motivation for this change? Is it a problem to build the tools? 
There is one problem I have with the change. You cannot build just with cmake anymore and it
forces the packages folder to be in the same directory tree as the binaries. From the inception of the CoreCLR, we were trying to ensure that you can build everything just using cmake. Recently, the eventing change has broken that by generating some headers / code in the build.cmd / sh, but we are in a process of fixing that.
So I would prefer making the msbuild invocation (including calling the vsdevcmd, which would mean adding a small cmd file that invokes the vsdevcmd and then the msbuild) the first COMMAND in the existing `add_custom_command(TARGET coreclr` and also placing the packages folder relative to the ${CMAKE_CURRENT_BINARY_DIR} if there is nothing preventing that. 
With these changes, plain cmake can be used to build everything.

**Comment by rahku** [comment]
There are multiple problems:
1. these tools do not change much and hence do not need to build everytime on everyone's machine
2. for arm64 builds these tools need to built as x64 binaries as they need to run during build

The nuget package needs to be restored only once on the machine and after that your cmake workflow should work just fine everytime....no need to invoke build.cmd. 

Well everything currently does not build using just cmake....mscorlib is built outside of cmake. This and other build tools package are currently already getting restored during mscorlib build. I just moved this one before native build. But I do still like your idea and will give it a try.

**Comment by rahku** [comment]
If I move running buildtools.proj as a custom command during native build ...I am getting the following error:
The OutputPath property is not set for project 'buildtools.proj'.  Please check to make sure that you have specified a valid combination of Configuration and Platform for this project.  Configuration='Debug'  Platform='amd64'.  You may be seeing this message because you are trying to build a project without a solution file, and have specified a non-default Configuration or Platform that doesn't exist for this project. [F:\coreclrGit\coreclr\src\scripts\buildtools\buildtools.proj]

@AlexGhiondea Is it possible to run buildtools.proj (managed build) during native compilation. Even though I do call VsDevCmd.bat before calling "msbuild buildtools.proj" ...it seems native compilation environment is interfering.

**Comment by janvorli** [comment]
@rahku Just a thought - do you have the VsDevCmd.bat call and the msbuilds call in the same .bat or .cmd file? 
And what should be the correct Configuration and Platform values?

**Comment by rahku** [comment]
passing outputPath parameter to msbuild ...makes it work. Will update the PR.

**Comment by rahku** [comment]
@janvorli Done changes as you suggested. Except
>> placing the packages folder relative to the ${CMAKE_CURRENT_BINARY_DIR} if there is nothing preventing that. 

mscorlib build is already using the packages folder wrt to root dir . So this does not change that.

**Comment by janvorli** [comment]
@rahku Ok. The fact that we are putting stuff relative to the root dir is unfortunate, since that prevents me from having the source folder mapped read only from my windows machine, which is the way I've always used. But I agree that it doesn't make sense to change it here when we already have it for mscorlib that way.

File: src/dlls/mscoree/coreclr/CMakeLists.txt [hunk 1]
```diff
@@ -130,38 +130,28 @@ if(WIN32)
...
-            COMMAND $<TARGET_FILE:GenClrDebugResource> /dac:$<TARGET_FILE:mscordaccore> /dbi:$<TARGET_FILE:mscordbi> /sku:onecoreclr /out:${CMAKE_CURRENT_BINARY_DIR}/clrDebugResource.bin
-            COMMAND $<TARGET_FILE:InjectResource> /bin:${CMAKE_CURRENT_BINARY_DIR}/clrDebugResource.bin /dll:$<TARGET_FILE:coreclr> /name:CLRDEBUGINFO
-            COMMENT Add dactable & debug resources to coreclr
-        )
-       
-    endif(CLR_CMAKE_PLATFORM_ARCH_ARM64)
- 
+    add_custom_command(TARGET coreclr
+        POST_BUILD
+        COMMAND ${CLR_DIR}/src/scripts/buildtools/buildtools.bat
```
Comment (janvorli) [inline src/dlls/mscoree/coreclr/CMakeLists.txt]: Just a tiny nit - naming the file restorebuildtools.bat would feel a bit easier to recognize what it does.

**Comment by janvorli** [comment]
LGTM

**Pull request merged**